app/controllers/loans_controller.rb

```diff
@@ -0,0 +1,33 @@
+class LoansController <  ApplicationController
+  before_action :set_loan, only: [:show]
+  def new
+    @loan = Loan.new
+  end
+
+  def create
+    @loan = Loan.create(loan_params)
+    if @loan.save
+      redirect_to @loan
+    end
+  end
+
+  def show
+    @loan = Loan.new
```

Comment:
Why would we need both `@loan` and `@seted_loan` here?

Reply:
The view loans/show is rendering a partial form and the calculated payments based on the id that was created just before. So `@loan` is needed for the model instance and `@seted_loan` is needed to get the information about the created loan.